util/hostname/match.go

```diff
@@ -0,0 +1,35 @@
+package hostname
+
+import "strings"
+
+// Match retruns true when the hostname specify the service passed by param.
+// It accepts the following hostname formats:
+// reviews, reviews.bookinfo.svc, reviews.bookinfo.svc.cluster.local,
+// *.bookinfo.svc, *.bookinfo.svc.cluster.local
+func MatchService(hostname, service, namespace string) bool {
+	domainParts := strings.Split(hostname, ".")
+	match := false
+
+	if len(domainParts) > 1 {
+		// hostname is a FQDN/Wildcard (e.g. reviews.bookinfo.svc, *.bookinfo.svc)
+		match = matchNamespace(domainParts, namespace, "svc")
+		if match && domainParts[0] != "*" {
+			match = domainParts[0] == service
+		}
+	} else {
+		// hostname is a service name (e.g. reviews)
+		match = hostname == service
+	}
+
+	return match
+}
+
+func matchNamespace(domain []string, namespace, resourceType string) bool {
```

Comment:
This is a one-liner function, as match is defined in any case, so I find strange to extract one-liner logic outside, but it's style, not a blocker, but I would extract if you would perform more than a single comparison. IMO.

Reply:
Also, "resourceType" is not exactly correct term, as "svc" is a suffix used in the custom Istio FQN, but it is not a type.